### PR TITLE
Clean description and docs CI

### PR DESCRIPTION
**Pull request by etiennebacher**
Several packages are no longer needed after #186

### CODE REVIEW

**Comment by etiennebacher** [comment]
R CMD check spends just one minute or less on `setup-r-dependencies` while the docs CI can spend 15 min on this because it rebuilds `polars` every time. It would be good to fix this before merging.

File: DESCRIPTION [hunk 1]
```diff
@@ -14,30 +14,27 @@ RoxygenNote: 7.2.3.9000
...
     bit64,
+    data.table,
+    devtools,
     knitr,
-    tibble,
+    nanoarrow,
+    nycflights13,
+    patrick,
     pillar,
+    rextendr,
```
[the remark this answers is not in the document]
Comment (etiennebacher) [reply DESCRIPTION:27]: I didn't know this field, can we put `testthat` there as well?

[the remark this answers is not in the document]
Comment (eitsupi) [reply DESCRIPTION:27]: I think they need to be in `Suggests` because they are needed for testing.

[the remark this answers is not in the document]
Comment (eitsupi) [reply DESCRIPTION:27]: In other words, CRAN ignores `Config/...` fields, so these are for only development.

[the remark this answers is not in the document]
Comment (etiennebacher) [reply DESCRIPTION]: done

**Comment by etiennebacher** [comment]
> R CMD check spends just one minute or less on `setup-r-dependencies` while the docs CI can spend 15 min on this because it rebuilds `polars` every time. It would be good to fix this before merging.

I can't find the reason why `r-lib/actions/setup-r-dependencies@v2` takes 15min for the docs CI but just 1 min for the check CI. There's probably some caching involved but I don't see the difference between the two workflows. @eitsupi do you know how to fix it?

**Comment by eitsupi** [comment]
I suspect that temporary directories are used when installing the local package. So I think the Rust binaries are not cached.

https://github.com/pola-rs/r-polars/blob/d6b2c64f990b7ddc33d75ddcb57102e3f6dc93a1/.github/workflows/docs.yaml#L95-L100

Perhaps it is worth simplifying the workflow by using the make command.

**Comment by etiennebacher** [comment]
We have to install the local package to run the vignettes properly so maybe there's no way to bypass that.

> Perhaps it is worth simplifying the workflow by using the make command.

I tried something like that but don't we lose some caching for packages to install? Feel free to push to the branch, I won't modify it more.

**Review by eitsupi — APPROVED** [review]
I think this workflow is good, thanks.

**Comment by eitsupi** [comment]
Obviously the Rust cache is still not working after merging this PR.
@sorhawell Any thoughts?

**Comment by eitsupi** [comment]
The cause was that the rust cache was configured before the rust-toolchain was configured.
I will fix in #219.

**Pull request merged**